node/src/store.rs

```diff
@@ -153,6 +154,12 @@ pub trait Store: Send + Sync + Debug {
     /// Remove header with given height from the store.
     async fn remove_height(&self, height: u64) -> Result<()>;
 
+    /// Set libp2p identity keypair for the store
+    async fn set_identity(&self, requested_keypair: Keypair) -> Result<()>;
```

Comment:
I think this should be `&mut` so that it can only be set before passing store to lumina

Reply:
folded setting the key into the store as discussed on daily.